instance/models/openedx_instance.py

```diff
@@ -314,24 +325,36 @@ def get_load_balancer_configuration(self, triggered_by_instance=False):
         The triggered_by_instance flag indicates whether the reconfiguration was initiated by this
         instance, in which case we log additional information.
         """
-        if not self.active_appserver:
+        active_appservers = self.get_active_appservers()
+        if not active_appservers.count():
             return self.get_preliminary_page_config(self.ref.pk)
-        ip_address = self.active_appserver.server.public_ip
-        if not ip_address:
+
+        # Create the haproxy backend configuration from the list of active appservers
+        template_vars = dict(
+            domain=self.domain,
+            http_auth_info_base64=self.http_auth_info_base64(),
+            appservers=[],
+        )
+        for appserver in active_appservers:
+            server_name = "appserver-{}".format(appserver.pk)
+            ip_address = appserver.server.public_ip
+            if ip_address:
+                template_vars['appservers'].append(dict(ip_address=ip_address, name=server_name))
+            else:
+                appserver.logger.error(
+                    "Active appserver does not have a public IP address. This should not happen."
+                )
+
+        if len(template_vars['appservers']) == 0:
             self.logger.error(
-                "Active appserver does not have a public IP address. This should not happen. "
+                "No active appservers found with public IP addresses.  This should not happen. "
                 "Deconfiguring the load balancer backend."
             )
             return [], []
-        backend_name = "be-{}".format(self.active_appserver.server.name)
-        server_name = "appserver-{}".format(self.active_appserver.pk)
+
+        backend_name = "be-{}".format(self.domain_slug)
```

Comment:
@smarnach Nice that backend_name was the only bit of appserver-specific information in the haproxy config too.. easy to factor out!